### DISCUSSION TITLE
rsHRF to estimate subject HRF in TVB using cortical surface

### DISCUSSION THREAD

**Discussion by michefabb** [compneuro-da/rsHRF#63] [General]
Dear experts,

I saw your post on TVB User forum at [https://groups.google.com/g/tvb-users/c/clPvlfB9X78/m/pfYD9mE5AwAJ](https://groups.google.com/g/tvb-users/c/clPvlfB9X78/m/pfYD9mE5AwAJ) and I am wondering how I can use rsHRF tool to estimate the subject HRF in each vertex of the cortical surface plus the subcortical structures (modeled as nodes) and then apply it in TVB: I am thinking to something similar to what is done for EEG when calculating the subject Leadfield Matrix and then applying it to generate the EEG signal.

I would appreciate your kind support!

Regards,

Mic

**Comment by danielemarinazzo** [comment]
Hi
whenever you have a time series of BOLD activity (voxel, vertex, region) you can estimate the rsHRF specific to that time series.
This rsHRF must then be upsampled to the sampling rate at which you simulate synaptic activity in TVB, and convolved with the simulated neural time series (_in lieu_ of the Balloon-Windkessel model). Then you will downsample the convolved time series to match the TR of your empirical BOLD.

Now, how to simulate vertex based synaptic activity in TVB (computational feasibility as well as way to do it) is a question for the TVB forum.

Good luck!

**Comment by michefabb** [comment]
Dear Prof. @danielemarinazzo ,

first of all let me thank you for your quick and kind reply.
The TVB surface simulation is ready and can generate the simulated BOLD, however this is done using the TVB default HRF (as you mentioned FirstOrderVolterra). I saw at your post on TVB Users [https://groups.google.com/g/tvb-users/c/clPvlfB9X78/m/pfYD9mE5AwAJ](https://groups.google.com/g/tvb-users/c/clPvlfB9X78/m/pfYD9mE5AwAJ) it is also referred how to change the TVB monitors.py [https://github.com/the-virtual-brain/tvb-root/pull/172/commits/f0a298bbc77df57800d6596a89b03180d623dd4d](https://github.com/the-virtual-brain/tvb-root/pull/172/commits/f0a298bbc77df57800d6596a89b03180d623dd4d) (by the way let me say you did a great job!).
At the same time a couple of points come to my mind:
- I am not able to use the python version on macOS (issue with installing `pyyawt` module) so I am using the docker (no GUI available for this), however I am unable to figure out how to use it to estimate the parameters of the HRF by command line with a surface...is there any guide? I was unable to find it in the instructions at [https://bids-apps.neuroimaging.io/rsHRF/](https://bids-apps.neuroimaging.io/rsHRF/)
- is there any fitting procedure included in the toolbox to check which of the six possible HRF equations fit better?

Appreciated!

Mic

**Reply by danielemarinazzo** [reply]
Hi

some replies here, then if the issue is specific to the python version, better move the conversation there

- the python version allows .gii and .gii.gz files as input, but not the CIFTI ones if you are using HCP data. It should be possible to do so with some work, maybe add a feature request, it would be great if someone could pick it up. Alternatively you can estimate the HRF with the matlab version and then use it in TVB.
- you mean comparing the several basis function options? You can check the sum of residuals (res_sum variable, both in python and matlab). Not sure there's a script that checks this automatically.

Tagging @NigelCol for the python implementation (better transfer the discussion there) and @guorongwu for the comparison of the fitting with different basis functions.

**Reply by michefabb** [reply]
Dear @danielemarinazzo ,

thanks again for your help. I am using data of a subject that does not come from HCP (I am using the HCPMMP1 atlas if this is what you meant). It looks the Python version uses a package (pyyawt) that cannot be installed on macOS with Apple Silicon processors (M1 M2 M3), making it impossible to use it (I can open a case on the proper github).
I could use the Matlab version, I read the User Manual for this, but it looks to refer more to the SPM plugin if I understood properly; just some doubt about the Matlab one:
- is it possible to install and use it without SPM installed?
- does it have a GUI, or this is only for the SPM plugin?
- are there specific commands to use a surface as input (or could you point me towards where I can find it)?

Appreciated!

Mic

**Reply by danielemarinazzo** [reply]
Which format is your surface data then?

- SPM (as rsHRF) does not need to be "installed", it's added to the path. Some SPM functions are used.
- the GUI is indeed the SPM plugin
- surface as input from command line is indicated here https://github.com/compneuro-da/rsHRF/blob/b86b2a1362c64baeaee36e8171041eb422354c31/rsHRF.m#L118. There is also a video in the readme https://www.bilibili.com/video/BV1z44y1b76m?spm_id_from=333.999.0.0

**Reply by michefabb** [reply]
Thanks again @danielemarinazzo ! The surface is in TVB format (3 .txt files for triangles, vertices, normals zipped in a unique file), they were created from the Freesurfer surfaces in Fsaverage5 (midthickness surface created as average between white and pial surfaces then downsampled to Fsaverage5) however I can save it as .gii using nibabel.gifti

I will have a look at the information you kindly provided and eventually revert.

Appreciated!

Mic
